Extend the chinese.lng to support Java record type

phoeagon opened the pull request:
(It doesn't really change much on the actual UI though. the texts are overridden in language definition files)

<img width="335" height="328" alt="Screenshot (1)" src="https://github.com/user-attachments/assets/cfa5745a-8f26-4b83-8bcd-ec4d77368ff1" />

```
extend chinese.lng file to translate for record (Java).
also updated the original "record".
"record" now reads "struct (Pascal: record)", using the more generally accepted term "struct" (in most other programming languages)
"custom" now reads "record (Java immutable data class)".

minor fixes in other places as well.
```

The situation with "struct" and "record" is a bit weird. Most books in Chinese will go with "struct" and Pascal is no longer a popular starter language.

(Note that the text above looks a bit weird, with uneven boldness/sizes. This is because I'm currently on a Windows 11 machine with locale set to "United States". This is due to legacy apps from Delphi using system fallback CJK fonts. AppLocale is no longer available from Microsoft, though other implementations still work)

phoeagon committed:
Update chinese.lng

phoeagon committed:
Update chinese.lng

phoeagon committed:
Merge branch 'albireo77:master' into master

phoeagon committed:
Update chinese.lng

phoeagon committed:
Merge branch 'albireo77:master' into master

phoeagon committed:
extend chinese.lng file to translate for record (Java).

also updated the original "record".
"record" now reads "struct (Pascal: record)", using the more generally accepted term "struct" (in most other programming languages)
"custom" now reads "record (Java immutable data class)".

minor fixes in other places as well.

albireo77 commented:
About these 2:
dtRecord=结构（Pascal：记录）
dtCustom=记录（Java只读数据类）
Sorry but there shouldn't be any specific programming language labels in translation file. If needed, these can be overriden by RecordLabel and CustomLabel values from language definition XML file.

phoeagon commented:
Hi! Thanks for the context!
Do you have a strong opinion that it should stick to Pascal terminology? Do you prefer to stick to the same terminology as "custom" instead of "record" (though none of these will be shown in the application due to the language definition file). Or is there a different section in the language definition file that could provide per-language labels I should look at please?
Thanks!

albireo77 commented:
devF UI is designed not to stick to any specific programming language though Pascal was the first programming language it has supported. Custom data type (dtCustom in https://github.com/albireo77/devflowcharter/blob/master/Common/Types.pas#L48) is only abstraction with intention to be labelled with real data type in given programming language via CustomLabel. As it was introduced just now to support records in Java the question you might have is why it isn't dtRecord. The answer is that dtRecord has already been used for a very long time (even when there were no records in Java) to support POJO classes in Java. And it must stay as it's not to break backward compatibility in saved project files.
The per-language labels you're asking are:
- RecordLabel
- CustomLabel
- EnumLabel
- OtherLabel

Declared here in reference XML file https://github.com/albireo77/devflowcharter/blob/master/LanguageDefinitions/Example.xml#L909.

phoeagon commented:
Hi!

Thanks for the info.

Yes I've seen the labels (`RecordLabel`, `CustomLabel`). I'm wondering if we plan to stick to a single label in the same programming language, across different natural languages (English, French, Chinese, etc) for these labels, or do we have any plans/mechanisms to localize these as well?

I can update the translation file to keep the existing naming and everything. My question is more about, given the current setup, my changes don't really show anything, because the only language with "custom" type enabled is Java, and it explicitly overrides this label to "record" (which is shared across English, French, etc).

Best,

albireo77 commented:
@phoeagon could you please update PR with latest changes in master branch? I mean translating entries EditorHelp29..32 added in https://github.com/albireo77/devflowcharter/commit/79cb82d94bdb72ccec8f69b4e77614e829ce0c40#diff-2796f3b84bfef37df64fda6e79f7c6ae5cef5864ab026f24fa66d31cfe7efffe. If you also rollback changes for entries dtRecord and dtCustom I will approve PR.

phoeagon committed:
Merge branch 'albireo77:master' into master

phoeagon committed:
updated chinese.lng

translated new help text.
reverted back translation of dtRecord and dtCustom to remove specific language contexts, to retain the existing naming paradigm

phoeagon committed:
remove extra whitespaces at the end of chinese.lng

phoeagon commented:
updated translation for the new help text, and reverted back the translation for dtCustom and dtRecord. dtCustom now reads "custom data"

albireo77 commented:
Note that I introduced numbering from EditorHelp23 entry - https://github.com/albireo77/devflowcharter/blob/master/translations/english.lng#L552. I am not sure if it looks ok in Chinese if entries 23, 25 and 27 have no numbers and 29, 31 have. Please verify. If you think that Chinese translation looks better without numbering, you can remove it.

phoeagon commented:
Thanks! Do you have an English screenshot that I can compare with please? I'll update it and load it in the system to verify this.

albireo77 commented:
Here it is:
<img width="427" height="206" alt="image" src="https://github.com/user-attachments/assets/1d46a75a-43b4-4335-8ca4-c456589baac8" />

phoeagon committed:
lite changes to chinese.lng. added numeric prefix to help text

phoeagon commented:
Thanks! Just updated the file with the numbered bullet point.

Also I just noticed that in the setting menu, there doesn't seem to be a translation label for "Auto-select code for highlighted flowchart elements". I can't find this in the english.lng file either.

albireo77 commented:
Yes, Related checkbox is removed in next version.
